scripts/us_census/pep/us_pep_sex/process.py

```diff
@@ -879,74 +1303,216 @@ def process(self):
+                pass
             # Taking the File name out of the complete file address
             # Used -1 to pickup the last part which is file name
             # Read till -4 inorder to remove the .tsv extension
+
             file_name = file_path.split("/")[-1][:-7]
             file_to_function_mapping = {
                 "pe-11-1": _national_1900_1979,
                 "us-est90int": _national_1990_2000,
+                "us-est90int-": _national_1990_2000,
                 "us-est00int": _national_2000_2010,
+                "us-est00int-": _national_2000_2010,
                 "nc-est2020-agesex-": _national_2010_2020,
-                "nc-est2021-agesex-": _national_2021,
+                "nc-est2020-agesex-r": _national_2010_2020,
+                "nc-est2023-agesex-": _national_2023,
+                "nc-est2023-agesex-r": _national_2023,
+                "nc-est2023-agesex-": _national_2023,
                 "pe": _state_1970_1980,
                 "stiag": _state_1980_1990,
+                "st-est00int-02-": _state_2000_2010,
                 "st-est00int-02": _state_2000_2010,
                 "SC-EST2020-AGESEX": _state_2010_2020,
-                "sc-est2021-syasex-01%2": _state_2021,
-                "sc-est2021-syasex-": _state_2021,
+                "SC-EST2020-AGESEX-": _state_2010_2020,
+                "sc-est2023-syasex-2": _state_2023,
+                "sc-est2023-syasex-3": _state_2023,
+                "sc-est2023-syasex-4": _state_2023,
+                "sc-est2023-syasex-5": _state_2023,
+                "sc-est2023-syasex-0": _state_2023,
+                "sc-est2023-syasex-1": _state_2023,
+                "sc-est2023-syasex-": _state_2023,
+                "sc-est2023-agesex-": _state_2023,
                 "co-asr-1": _county_1970_1980,
                 "pe-02-1": _county_1980_1990,
                 "stch-icen1": _county_1990_2000,
                 "co-est00int-agesex-": _county_2000_2010,
+                "co-est00int-agesex-5": _county_2000_2010,
                 "CC-EST2020-AGESEX-": _county_2010_2020,
-                "cc-est2021-agesex-": _county_2021
+                "CC-EST2020-AGESEX-A": _county_2010_2020,
+                "cc-est2023-agesex-": _county_2023,
+                "cc-est2023-agesex-a": _county_2023,
+                "nc-est2023-agesex-": _national_2023,
+                "nc-est2023-agesex-r": _national_2023,
+                "sc-est2023-syasex-2": _state_2023,
+                "sc-est2023-syasex-3": _state_2023,
+                "sc-est2023-syasex-4": _state_2023,
+                "sc-est2023-syasex-5": _state_2023,
+                "sc-est2023-syasex-0": _state_2023,
+                "sc-est2023-syasex-1": _state_2023,
+                "sc-est2023-syasex-": _state_2023,
+                "sc-est2023-agesex-": _state_2023,
+                "cc-est2023-agesex-": _county_2023,
+                "cc-est2023-agesex-a": _county_2023
```

Comment:
can we also extend to handle future years assuming the dame format?

Reply:
Yes modified accordingly